tests/conftest.py

```diff
@@ -10,122 +11,235 @@
+        user.id = self.unique_id()
+        user.discriminator = discriminator
 
-    def _factory(guild, name='Mock Member', discriminator='1234'):
-        member = mocker.Mock(spec=discord.Member)
-        member.id = make_discord_id()
-        member.guild = guild
-        member.name = name
-        member.discriminator = discriminator
-        return member
+        if guild is None:
+            user.name = name or 'Mock User'
+        else:
+            user.name = name or 'Mock Member'
+            user.guild = guild
 
-    return _factory
+        return user
 
+    def make_text_channel(self, guild, name='mock-channel'):
+        """Returns a new TextChannel-like Mock."""
 
-@pytest.fixture
-def make_discord_text_channel(mocker, make_discord_id):
-    """Factory fixture to generate discord.TextChannel-like Mock objects."""
-
-    def _factory(guild, name='mock-channel'):
-        channel = mocker.Mock(spec=discord.TextChannel)
-        channel.id = make_discord_id()
+        channel = self.pytest_mocker.Mock(
+            spec=discord.TextChannel, name='channel')
+        channel.id = self.unique_id()
         channel.guild = guild
         channel.type = discord.ChannelType.text
         channel.name = name
         channel.topic = f'topic for {name}'
+        channel.test_history = []  # The messages that history() will return.
+
+        async def get_history(limit=100, oldest_first=None):
+            limit = len(channel.test_history) if limit is None else limit
+            history = channel.test_history
+            slice = history[:limit] if oldest_first else history[:-limit:-1]
+            for m in slice:
+                yield m
+
+        async def fetch_message(id):
+            found = next((m for m in channel.test_history if m.id == id), None)
+            if found:
+                return found
+            else:
+                raise discord.NotFound(None, f'message {id} not found')
+
+        channel.history.side_effect = get_history
+        channel.fetch_message.side_effect = fetch_message
         return channel
 
-    return _factory
+    def make_message(self, guild, channel, author, content='Mock content'):
+        """Returns a new Message-like Mock."""
 
-
-@pytest.fixture
-def make_discord_message(mocker, make_discord_id):
-    """Factory fixture to generate discord.Message-like Mock objects."""
-
-    def _factory(guild, channel, author, content='Mock message content'):
-        message = mocker.Mock(spec=discord.Message)
-        message.id = make_discord_id()
+        message = self.pytest_mocker.Mock(spec=discord.Message, name='message')
+        message.id = self.unique_id()
         message.guild = guild
         message.channel = channel
         message.author = author
         message.content = content
+        message.reactions = []
         return message
 
-    return _factory
+    #
+    # Helper methods to update data in useful ways.
```

Comment:
This comment makes me wonder if you are planning to add helpers to update data in useless ways. 😹

Reply:
Oh boy am I!